### PR TITLE
Allow unhandled subscriber exceptions to be thrown

### PR DESCRIPTION
**Pull request by rasmus**
Sometimes unhandled subscribers exceptions should be allowed to propagate to the originating publisher of a command.

### CODE REVIEW

File: Source/EventFlow/Subscribers/DispatchToEventSubscribers.cs [hunk 1]
```diff
@@ -83,8 +105,8 @@ private class SubscriberInfomation
             {
                 _log.Error(
                     exception,
-                    "Failed to dispatch to event handler {0}",
-                    handler.GetType().PrettyPrint());
+                    $"Failed to dispatch to event handler '{handler.GetType().PrettyPrint()}' for event '{domainEvent.EventType.PrettyPrint()}' due to unexpected exception!");
+                throw;
```
Comment (ylbillyli) [inline Source/EventFlow/Subscribers/DispatchToEventSubscribers.cs]: How about save one try-catch overhead by doing it here:
```
if(_eventFlowConfiguration.ThrowSubscriberExceptions) throw else _log.debug(e, "swallowed");
```
Just a small suggestion, not sure if I am up-to-date with the code base, if you are doing an extra `try-catch` in `DomainEventPublisher` on purpose to make it neater, then I think that is also fair enough.

Comment (rasmus) [inline Source/EventFlow/Subscribers/DispatchToEventSubscribers.cs]: @ylbillyli You are right. I was thinking that I wanted to preserve the previous logging behavior, but a cleaner codebase is more favorable

**Comment by rasmus** [comment]
@ylbillyli I did some changes based on your comments. It cleaned it up a bit

**Comment by ylbillyli** [comment]
@rasmus looks perfect to me

**Pull request merged**